doc/nrf/libraries/others/emds.rst

```diff
@@ -0,0 +1,132 @@
+
+The :c:func:`emds_is_ready` function can be called to check if EMDS is prepared to store the data.
+
+Once the data storage has completed, a callback is called if provided in :c:func:`emds_init`.
+This callback notifies the application that the data storage has completed, and can be used to reboot the CPU or execute another function that is needed.
+
+After completion of :c:func:`emds_store`, the :c:func:`emds_is_ready` function call will return error, since it can no longer guarantee that the data will fit into the flash area.
+
+The above described process is summarized in a message sequence diagram.
+
+.. msc::
+    hscale = "1.3";
+    Application, EMDS;
+    --- [ label = "Application initialization started" ];
+    Application=>EMDS         [ label = "emds_init(emds_store_cb_t)" ];
+    --- [ label = "Initialization of all functionality that does emds_entry_add()" ];
+    Application=>EMDS         [ label = "emds_entry_add(1)" ];
+    Application=>EMDS         [ label = "emds_entry_add(2)" ];
+    ...;
+    Application=>EMDS         [ label = "emds_entry_add(n)" ];
+    --- [ label = "All emds_entry_add() executed" ];
+    Application=>EMDS         [ label = "emds_load()" ];
+    Application=>EMDS         [ label = "emds_prepare()" ];
+    --- [ label = "Application initialization ended" ];
+    ...;
+    Application->Application  [ label = "Interrupt calling emds_store()" ];
+    Application=>EMDS         [ label = "emds_store()" ];
+    EMDS box EMDS [ label = "Thread storing data executing" ];
+    Application<<=EMDS        [ label = "emds_store_cb_t callback" ];
+    Application->Application [ label = "Reboot/halt" ];
+
+Requirements
+************
+To prevent frequent writes to flash memory, the EMDS library can write data to flash only when the device is shutting down.
+EMDS restores the application data to RAM at reboot.
+
+EMDS can store data within a guaranteed time, based on the amount of data being stored.
+EMDS can be used to store data in memory in situations of critical power shortage, for example before the device battery is depleted.
+It is important that the hardware has the appropriate functionality to sustain power long enough for the storage to be completed before the device is fully discharged.
+
+Configuration
+*************
+To initialize the emergency data storage, completed the following steps:
+
+1. Enable the :kconfig:option:`CONFIG_EMDS` Kconfig option.
+#. Include the :file:`emds/emds.h` file in your :file:`main.c` file.
+#. Create the callback function :c:func:`emds_store_cb_t`.
+#. Call the :c:func:`emds_init` function.
+#. Call init of all functions through :c:func:`emds_entry_add`.
+#. Call :c:func:`emds_load`.
+#. Call :c:func:`emds_prepare`.
+#. Create interrupt or other functionality that will call :c:func:`emds_store`.
+
+Limitations
+***********
+.. note::
```

Comment:
This does not need to be a note. It is placed under limitations, it should be sufficient :)

Reply:
Also, replace 'this feature' with EMDS :)